File: node/zkgroup/profiles/ProfileKeyCredential.ts
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray';
import NativeImpl from '../../NativeImpl';

export default class ProfileKeyCredential extends ByteArray {

  static SIZE = 145;

  constructor(contents: Buffer) {
    super(contents, ProfileKeyCredential.SIZE, true);
    NativeImpl.ProfileKeyCredential_CheckValidContents(contents);
  }
}

File: java/java/src/main/java/org/signal/zkgroup/profiles/ProfileKeyCredential.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.profiles;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class ProfileKeyCredential extends ByteArray {
  public ProfileKeyCredential(byte[] contents) throws InvalidInputException {
    super(contents);
    try {
      Native.ProfileKeyCredential_CheckValidContents(contents);
    } catch (IllegalArgumentException e) {
      throw new InvalidInputException(e.getMessage());
    }
  }
}
